specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2024-02-15-preview/deploymentSettings.json

```diff
@@ -0,0 +1,974 @@
+      }
+    },
+    "SdnIntegration": {
+      "type": "object",
+      "description": "SDN Integration config to deploy AzureStackHCI Cluster.",
+      "properties": {
+        "networkController": {
+          "$ref": "#/definitions/NetworkController",
+          "description": "network controller config for SDN Integration to deploy AzureStackHCI Cluster."
+        }
+      }
+    },
+    "DeploymentSecuritySettings": {
+      "type": "object",
+      "description": "The SecuritySettings of AzureStackHCI Cluster.",
+      "properties": {
+        "hvciProtection": {
+          "type": "boolean",
+          "description": "By default, Hypervisor-protected Code Integrity is enabled on your Azure HCI cluster.",
+          "default": true
+        },
+        "drtmProtection": {
+          "type": "boolean",
+          "description": "By default, Secure Boot is enabled on your Azure HCI cluster. This setting is hardware dependent.",
+          "default": true
+        },
+        "driftControlEnforced": {
+          "type": "boolean",
+          "description": "When set to true, the security baseline is re-applied regularly.",
+          "default": true
+        },
+        "credentialGuardEnforced": {
+          "type": "boolean",
+          "description": "When set to true, Credential Guard is enabled.",
+          "default": false
+        },
+        "smbSigningEnforced": {
+          "type": "boolean",
+          "description": "When set to true, the SMB default instance requires sign in for the client and server services.",
+          "default": true
+        },
+        "smbClusterEncryption": {
+          "type": "boolean",
+          "description": "When set to true, cluster east-west traffic is encrypted.",
+          "default": false
+        },
+        "sideChannelMitigationEnforced": {
+          "type": "boolean",
+          "description": "When set to true, all the side channel mitigations are enabled",
+          "default": true
+        },
+        "bitlockerBootVolume": {
+          "type": "boolean",
+          "description": "When set to true, BitLocker XTS_AES 256-bit encryption is enabled for all data-at-rest on the OS volume of your Azure Stack HCI cluster. This setting is TPM-hardware dependent. ",
+          "default": true
+        },
+        "bitlockerDataVolumes": {
+          "type": "boolean",
+          "description": "When set to true, BitLocker XTS-AES 256-bit encryption is enabled for all data-at-rest on your Azure Stack HCI cluster shared volumes.",
+          "default": true
+        },
+        "wdacEnforced": {
+          "type": "boolean",
+          "description": "WDAC is enabled by default and limits the applications and the code that you can run on your Azure Stack HCI cluster.",
+          "default": true
+        }
+      }
+    },
+    "Storage": {
+      "type": "object",
+      "description": "The Storage config of AzureStackHCI Cluster.",
+      "properties": {
+        "configurationMode": {
```

Comment:
Consider model it as an enums if the property has a finite set of values?
refs: https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi015-avoid-modelling-a-property-as-string-type-when-it-should-be-an-enums

Reply:
These model is derived from HCI cluster deployment Answerfile which customer are already using and want to use same model with ARM. Used same data type used by customers so can't modify it now. Also this property already exposed as GA API https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2024-01-01/deploymentSettings.json